lib/galaxy/workflow/modules.py

```diff
@@ -697,6 +700,59 @@ def get_runtime_inputs(self, **kwds):
         # TODO: Use a dict-based description from YAML tool source
         element = Element("param", name="input", label=self.label, type=parameter_type, optional=str(optional), **parameter_kwds)
         input = parameter_class(None, element)
+
+        # Retrieve possible runtime options for 'select' type inputs
+        if parameter_type == 'select' and connections:
+            static_options = []
+            for connection in connections:
+                tool_inputs = connection.input_step.module.tool.inputs
+                # Account for select options within conditionals
+                if connection.input_name not in tool_inputs and "|" in connection.input_name:
+                    param_name, case_name = connection.input_name.split("|", 1)
+                    if param_name in tool_inputs:
+                        param = tool_inputs[param_name]
+                    elif re.match('.+_\\d+', param_name):  # check for repeat label
+                        param_name, param_idx = param_name.rsplit('_', 1)
+                        param = tool_inputs[param_name]
+                    while param.type in ("conditional", "repeat"):
+                        if "|" in case_name:
+                            param_name, case_name = case_name.split("|", 1)
+                            if param.type == "conditional":
+                                sel_cases = [case for case in param.cases if param_name in case.inputs]
+                                if sel_cases:
+                                    sel_case = sel_cases[0]
+                                elif re.match('.+_\\d+', param_name):  # check for repeat label
```

Comment:
The regex pattern should be declared as a **raw** string:  **r**'.+_\d+'  
The repeat_param will be the repeat name which can be at least one character: '.+'
followed by the underscore character: '_'
followed by an integer repeat index: '\d+'
```suggestion
                                elif re.match(r'.+_\d+', param_name):  # check for repeat label
```